src/rhsmlib/facts/virt.py

```diff
@@ -148,6 +148,13 @@ def get_all(self):
         virt_what_info = virt_what_collector.get_all()
         virt_info.update(virt_what_info)
 
+        # Ensure we do not gather the virt.uuid fact for host_types we cannot
+        # See BZ#1438085
+        for no_uuid_host_type in VirtUuidCollector.no_uuid_platforms:
+            if virt_what_info.get('virt.host_type', None) is None or \
+               virt_what_info['virt.host_type'].find(no_uuid_host_type) > -1:
```

Comment:
Is there a reason to use `find`?  E.g. we're looking for a specific substring in `virt.host_type`?  Otherwise, it seems like `virt_what_info['virt.host_type'] in no_uuid_host_type` would be a more straightforward predicate.